aten/src/ATen/core/dispatch/OperatorEntry.h

```diff
@@ -157,9 +220,10 @@ class CAFFE2_API OperatorEntry final {
   // re-executed and then only allow one kernel here, i.e. error if a kernel
   // is already registered, but that's a lot of effort to implement and
   // currently not high-pri.
-  ska::flat_hash_map<c10::optional<DispatchKey>, std::list<KernelEntry>> kernels_;
+  ska::flat_hash_map<DispatchKey, std::list<AnnotatedKernel>> kernels_;
 
-  std::mutex kernelsMutex_; // protects kernels_
```

Comment:
Is removing this mutex safe?

Reply:
I forgot to write this in the PR description, but the change of the locking strategy is that there is now a single lock for all of Dispatcher and everything it owns, rather than per OperatorEntry locks. This is because precomputation needs to access state in Dispatcher.